src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -234,6 +236,39 @@ private static bool IsFirewallEnabled()
                 return false;
             }
         }
+
+        private async Task<string> GetPsVersionInfo()
+        {
+            var builder = new StringBuilder();
+
+            string powerShellExe = HostContext.GetService<IPowerShellExeUtil>().GetPath();
+            string arguments = @"Write-Host ($psversiontable | Out-String)";
+            using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
+            {
+                processInvoker.OutputDataReceived += (object sender, ProcessDataReceivedEventArgs args) =>
```

Comment:
i would just print both STDOUT/STDERR to output and not to `Trace`

Reply:
executionContext.Output?